lib/switch.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import { code, CodeExample, PropsDefinition, Prop } from './story-utils';
+
+const ShuttleBody = styled.span`
+  position: relative;
+  display: inline-block;
+  height: var(--space-3);
+  width: var(--space-5);
+  font-weight: bold;
+  font-size: var(--fontSizes-small);
+  transition: transform 0.3s;
+  transform: translateX(-40%);
+  transform: ${({ active }) => active && 'translateX(0%)'};
+`;
+
+const ShuttleLabel = styled.span`
+  display: inline-block;
+  width: var(--space-4);
+`;
+const OnLabel = styled(ShuttleLabel)`
+  color: var(--colors-success-text);
+  background-color: var(--colors-success-background);
+  text-align: right;
+  padding: var(--space-1) calc(var(--space-2) + var(--space-1)) var(--space-1) 0;
+`
+const OffLabel = styled(ShuttleLabel)`
+  color: var(--colors-tertiary-text);
+  background-color: var(--colors-tertiary-background);
+  text-align: left;
+  padding: var(--space-1) 0 var(--space-1) calc(var(--space-2) + var(--space-1));
+`
+
+const Knob = styled.span`
+  position: absolute;
+  top: 0;
+  left: 50%;
+  transform: translateX(-50%);
+  border-radius: var(--space-2);
+  background-color: var(--colors-background);
+  height: calc(var(--space-3) - 4px);
+  width: calc(var(--space-3) - 4px);
+  box-shadow: 0 0 4px rgba(0,0, 0,0.1);
+`;
+
+const Shuttle = (props) => (
+  <ShuttleBody {...props}>
+    <OnLabel>
+      On
+    </OnLabel>
+    <OffLabel>Off</OffLabel>
+    <Knob />
+  </ShuttleBody>
+);
+
+const SwitchBody = styled.label`
+  position: relative;
+  display: inline-block;
+  border-radius: var(--space-2);
+  border: 2px solid var(--colors-primary);
+  height: var(--space-3);
+  width: calc(var(--space-4) + var(--space-2));
+  vertical-align: middle;
+  overflow: hidden;
+  &:focus-within {
+    box-shadow 0 0 4px blue;
+  }
+`;
+
+const HiddenCheckbox = styled.input.attrs(() => ({ type: 'checkbox' }))`
+  position: absolute;
+  opacity: 0;
```

Comment:
With opacity set to 0 here it seems Voiceover won't pick up that you're on an input and so isn't able to toggle the switch at all. Is this the same mechanism we're using to hide checkboxes in other types of toggle components (like that lock)?

Reply:
Looks like it: https://github.com/FogCreek/Glitch-Community/blob/master/src/components/fields/hidden-checkbox.styl